### PR TITLE
Fix for Null Pointer Exception error in the editor created event.

### PR DESCRIPTION
**Pull request by ashishrp-aws**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

#4554
Fix for Null Pointer Exception error in the editor created event check.

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changes/next-release/bugfix-8ded9b7b-7f3a-45c8-a480-974e2d0ac4a6.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "Added fix to remove NullPointerException error for editor created event."
```
Comment (rli) [inline .changes/next-release/bugfix-8ded9b7b-7f3a-45c8-a480-974e2d0ac4a6.json:3]:
```suggestion
  "description" : "Resolve a NullPointerException that could occur while handling editor creation event."
```

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ import software.aws.toolkits.jetbrains.services.codewhisperer.util.CodeWhisperer
 internal class CodeWhispererCodeScanFileListener(val project: Project) : EditorFactoryListener {
     override fun editorCreated(event: EditorFactoryEvent) {
         val actionManager = CodeWhispererExplorerActionManager.getInstance()
-        if (event.editor.virtualFile.isFile && event.editor.project == project && actionManager.isAutoEnabledForCodeScan() &&
+        if ( event.editor.project == project && actionManager.isAutoEnabledForCodeScan() &&
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt:18]: don't we still need to check if it's a file?

Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt:18]: The check is added in later for file and validity of file.

**Review by Will-ShaoHua — APPROVED** [review]

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt [hunk 3]
```diff
@@ -15,7 +14,8 @@ import software.aws.toolkits.jetbrains.services.codewhisperer.util.CodeWhisperer
 internal class CodeWhispererCodeScanFileListener(val project: Project) : EditorFactoryListener {
     override fun editorCreated(event: EditorFactoryEvent) {
         val actionManager = CodeWhispererExplorerActionManager.getInstance()
-        if (event.editor.virtualFile.isFile && event.editor.project == project && actionManager.isAutoEnabledForCodeScan() &&
+        if (
+            event.editor.project == project && actionManager.isAutoEnabledForCodeScan() &&
```
Comment (Will-ShaoHua) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt:18]: nit 
```
a &&
b &&
c &&
d
```

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt [hunk 4]
```diff
@@ -15,7 +14,8 @@ import software.aws.toolkits.jetbrains.services.codewhisperer.util.CodeWhisperer
 internal class CodeWhispererCodeScanFileListener(val project: Project) : EditorFactoryListener {
     override fun editorCreated(event: EditorFactoryEvent) {
         val actionManager = CodeWhispererExplorerActionManager.getInstance()
-        if (event.editor.virtualFile.isFile && event.editor.project == project && actionManager.isAutoEnabledForCodeScan() &&
```
Comment (Will-ShaoHua) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt:18]: we could simply early return 
```
if (event.editor.virtualFile) {
  return
}
```

Comment (rli) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt:18]: given the amount of functionality that seems to indirectly need an editor, i think we should early return if the user didn't actually touch a file

**Review by rli — APPROVED** [review]
Please determine if outstanding comments are nonblocking or can be addressed later, and have an item to fast-follow with a test case for the issue being fixed

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt (hunk 4 above)
Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanFileListener.kt:18]: testing the early return now.

**Pull request merged**